gitwrapperlib/gitwrapperlib.py

```diff
@@ -123,16 +123,30 @@ def add_remote_origin(self, url):
         """Adds the remote origin."""
         self._git.remote('add', 'origin', url)
 
+    def push_default(self):
+        """Pushes to default branch."""
+        branch = self._git('symbolic-ref', '--short', 'refs/remotes/origin/HEAD').split('/')[-1].rstrip()
```

Comment:
This for me returns 
```
git symbolic-ref --short refs/remotes/origin/HEAD
fatal: ref refs/remotes/origin/HEAD is not a symbolic ref
```
on a random project. Is this something that is project specific to your project that would not work in a general way maybe?

Reply:
We could drop the push_default methods, however we do need a way to figure out what the default branch is.
It's pretty inconvenient to merge feature branches if there is no way to figure out what the default branch is.

My code was based on https://stackoverflow.com/questions/28666357/git-how-to-get-default-branch

Reply:
From what I see on that thread, first answer on first topic `The default branch is a github thing, not a git thing` . I think that we should not implement vendor specific implementations. I see different solutions for gitlab, github and I am sure other vendor solutions. This is quite tricky.

Reply:
I've added a slightly different version, which should work in most cases...